### PR TITLE
sf 17

### PR DESCRIPTION
**Pull request by niklasf**
pull request from @Disservin's branch

### CODE REVIEW

**Comment by Disservin** [comment]
btw what does the suffixes -40, -7, -70 even mean? My 80 was choosen rather randomly

**Comment by niklasf** [comment]
I think it's download size of NNUE, to distinguish the cases where we have multiple options.

**Comment by niklasf** [comment]
Added Stockfish 17 nets to lifat: https://github.com/lichess-org/lifat/commit/9df495f25a95118e42e0f9a34db6e08654142e72

File: patches/sf17-79.patch [hunk 1]
```diff
@@ -107,6 +107,19 @@ index 15208255..7f9f7dfa 100644
...
+diff --git a/src/timeman.h b/src/timeman.h
+index 10207a8a..403006a5 100644
+--- a/src/timeman.h
++++ b/src/timeman.h
+@@ -48,7 +48,7 @@ class TimeManagement {
+     TimePoint elapsed(FUNC nodes) const {
+         return useNodesTime ? TimePoint(nodes()) : elapsed_time();
+     }
+-    TimePoint elapsed_time() const { return now() - startTime; };
++    TimePoint elapsed_time() const { return std::max(now() - startTime, TimePoint(1)); };
```
Comment (Disservin) [inline patches/sf17-79.patch:119]: im curious about this change, why did you need that ? We are using `std::chrono::steady_clock` and it is supposed to be monotonic ? 
> Class std::chrono::steady_clock represents a monotonic clock. The time points of this clock cannot decrease as physical time moves forward and the time between ticks of this clock is constant.

Comment (niklasf) [reply patches/sf17-79.patch:119]: Unfortunately emscripten is failing to reliably provide monotonicity across threads in separate web workers.

* After https://github.com/emscripten-core/emscripten/issues/9783 it's roughly correct, at least
* https://github.com/emscripten-core/emscripten/issues/15380 is still open, though

Comment (niklasf) [reply patches/sf17-79.patch:119]: Changed to `TimePoint(0)`, which is more precise (5ccdf71971099190a351910d7a9df6c59a8f73db)

Comment (Disservin) [reply patches/sf17-79.patch:119]: Oh yikes, thanks...

Comment (schlawg) [reply patches/sf17-79.patch:119]: we do use PROXY_TO_PTHREAD, is it still an issue?

**Comment by schlawg** [comment]
outstanding. stockfish 17 is merged to lila/master and awaiting deploy

File: patches/sf17-79.patch (hunk 1 above)
Comment (niklasf) [reply patches/sf17-79.patch:119]: Unless there's been further patches, each worker asks the parent thread for an offset/skew when it's first created. That's rarely off by more than a few milliseconds, but still doesn't guarantee monotonicity in all cases (measure in parent after child, measure in child a after child b).

Proxying every single time call to a specific thread would work, but I don't think that was ever implemented.

**Comment by Disservin** [comment]
Yes, thank you both for working on the update :D I think this is the fastest stockfish update lichess ever got :)

**Comment by Disservin** [comment]
> outstanding. stockfish 17 is merged to lila/master and awaiting deploy

I see it on lichess but nginx doesnt see the network, I assume the assets were not properly updated? https://lichess1.org/assets/lifat/nnue/nn-1111cefa1111.nnue

**Comment by niklasf** [comment]
Yes, there's an issue with the deployment script that we have to look into. Meanwhile it was fixed manually.
